### PR TITLE
Add profile page

### PR DESCRIPTION
**Pull request by neokry**
## Description

Adds profile pages for users

## Motivation & context

- users want to show off dao membership
- users want to see what daos other users are in

## Code review

- does profile page correctly show all DAO tokens for a user
- does layout work properly on desktop and mobile
- activity page was refactored to share pagination logic, did this break anything?

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have done a self-review of my own code
- [x] Any new and existing tests pass locally with my changes
- [x] My changes generate no new warnings (lint warnings, console warnings, etc)

### CODE REVIEW

**Review by isaaczaak — CHANGES_REQUESTED** [review]
Add maxwidth to profile page so it lines up with universal navbar

**Comment by isaaczaak** [comment]
Follow-up PR:

Rearrange dropdown component so that this section is at the top, just underneath `Profile` and `Create a DAO` CTAs.
Screenshot 2023-05-22 at 4 53 53 pm

File: apps/web/src/data/graphql/requests/tokensQuery.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import { sdk } from '../client'
+import { SortDirection, TokenSortKey } from '../sdk.generated'
+
+export interface TokensQueryResponse {
+  tokens: {
+    tokenId: string
+    collection?: string
+    image?: string
+    name?: string
```
Comment (nadiakiamilev) [inline apps/web/src/data/graphql/requests/tokensQuery.ts:12]: i believe you can get the typed response from the generated sdk file

File: apps/web/src/pages/api/profile/[user]/tokens.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+export interface UserTokensResponse {
+  tokens?: TokensQueryResponse
+  daos: MyDaosResponse
+}
+
+export const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { user, page } = req.query
+  const baseUrl = getBaseUrl()
+
+  console.log('user', user)
```
Comment (nadiakiamilev) [inline apps/web/src/pages/api/profile/[user]/tokens.tsx:17]: nit: console statement

File: apps/web/src/pages/api/profile/[user]/tokens.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import axios from 'axios'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+import { MyDaosResponse } from 'src/data/graphql/requests/daoQuery'
+import { TokensQueryResponse, tokensQuery } from 'src/data/graphql/requests/tokensQuery'
+import { getBaseUrl } from 'src/utils/baseUrl'
+
+export interface UserTokensResponse {
```
Comment (nadiakiamilev) [inline apps/web/src/pages/api/profile/[user]/tokens.tsx:8]: it looks like this file is a tsx file even thought it's an api route. is that intended? I also noticed `api/profile/[user]/daos` is the same.

File: apps/web/src/pages/api/profile/[user]/tokens.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    })
+
+  const collections = daos?.map((dao) => dao.collectionAddress)
+
+  const tokens = await tokensQuery(
+    [user as string],
+    collections,
+    page ? parseInt(page as string) : undefined
+  )
+  res.status(200).json({ tokens, daos } as UserTokensResponse)
```
Comment (nadiakiamilev) [inline apps/web/src/pages/api/profile/[user]/tokens.tsx:35]: do we also need to handle some error cases? ie not found?

File: apps/web/src/pages/profile/[user].tsx [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        <Box
+          position={'fixed'}
+          h="100vh"
+          top="x0"
+          style={{ left: '27%', borderRight: '2px solid #F2F2F2' }}
+        />
+      )}
+
+      <Box
+        mt={isMobile ? 'x14' : undefined}
```
Comment (nadiakiamilev) [inline apps/web/src/pages/profile/[user].tsx:128]: nit: would prefer the built-in breakpoints here instead of isMobile

**Comment by neokry** [comment]
thanks for the review @nadiakiamilev just pushed a few commits that should resolve these issues you mentioned.

**Review by isaaczaak — APPROVED** [review]
tested

**Review by nadiakiamilev — APPROVED** [review]

**Pull request merged**